src/prommis/uky/costing/tests/test_ree_costing.py

```diff
@@ -5265,13 +5412,16 @@ def test_REE_costing_recovery_transportcostParamnounits():
         transport_cost_per_ton_product=m.fs.transport_cost_per_ton_product,
     )
 
+    dt = DiagnosticsToolbox(m)
```

Comment:
There appear to be a lot of very similar tests here - would it be possible to do this as a single parameterized test?

Reply:
I can parameterize parts of the file, particularly the parts that test different expressions, parameters, or variables for the same argument.

Is there a way to parameterize the entire file, or does too much differ between the tests?

Reply:
The parts where the tests are primarily the same (so the different expressions, parameters and variables, etc.). Some of the tests will be unique, but the point here is to minimize the amount of duplicated code that we need to maintain (which is a balancing act between having a small amount of highly re-used code that is hard to understand and maintain).

Reply:
Unfortunately the way GitHub parsed the code difference is not very useful for reviewing. If you look at the file itself, I combined the options for (land_cost, additional_chemical_cost, additional_waste_cost) on line 2749, the options for recovery_rate units on line 3931, and the options for transport_cost on line 4137.